pkg/api/v1/client.go

```diff
@@ -0,0 +1,128 @@
+package api
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// New creates a new API client that implements the CourierClient interface.
+func New(endpoint string, opts ...ClientOption) (_ CourierClient, err error) {
+	if endpoint == "" {
+		return nil, ErrEndpointRequired
+	}
+
+	// Create a client with the parsed endpoint.
+	c := &APIv1{}
+	if c.url, err = url.Parse(endpoint); err != nil {
+		return nil, err
+	}
+
+	// Apply options
+	for _, opt := range opts {
+		if err = opt(c); err != nil {
+			return nil, err
+		}
+	}
+
+	// If a client hasn't been specified, create the default client.
+	if c.client == nil {
+		c.client = &http.Client{
+			Transport:     nil,
+			CheckRedirect: nil,
+			Timeout:       30 * time.Second,
+		}
+	}
+	return c, nil
+}
+
+// APIv1 implements the CourierClient interface.
+type APIv1 struct {
+	url    *url.URL
+	client *http.Client
+}
+
+var _ CourierClient = &APIv1{}
+
+//===========================================================================
+// Client Methods
+//===========================================================================
+
+// Status returns the status of the courier service.
+func (c *APIv1) Status(ctx context.Context) (out *StatusReply, err error) {
```

Comment:
curious why the variable is named `c` instead of `s`

Reply:
I was thinking "c" for client.